### PR TITLE
Fix ReceiveTimeout issue with NotInfluenceReceiveTimeout #24770

### PR DESCRIPTION
**Pull request by newca12**

### CODE REVIEW

**Comment by newca12** [comment]
Fix #24770

**Comment by ktoso** [comment]
Thanks for the PR!

**Comment by ktoso** [comment]
OK TO TEST

File: akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala [hunk 1]
```diff
@@ -28,16 +28,13 @@ private[akka] trait ReceiveTimeout { this: ActorCell ⇒
 
   final def checkReceiveTimeout() {
     val recvtimeout = receiveTimeoutData
-    //Only reschedule if desired and there are currently no more messages to be processed
-    if (!mailbox.hasMessages) recvtimeout._1 match {
+    recvtimeout._1 match {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala]: I agree that this is the most straight forward way to implement it. I'm somewhat concerned about the performance concern of always rescheduling, but other solutions I can think of are too complicated and have other drawbacks.

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**